File: blockchain-info-aggregation-stream-parent/blockchain-info-aggregation-stream-client-example/src/main/java/info/blockchain/aggregation/client/Application.java
package info.blockchain.aggregation.client;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.reactive.socket.WebSocketMessage;
import org.springframework.web.reactive.socket.client.ReactorNettyWebSocketClient;
import reactor.core.scheduler.Schedulers;

import java.net.URI;

@Slf4j
@SpringBootApplication
public class Application implements ApplicationRunner {

    @Value("#{new java.net.URI('${websocket-server.uri}')}")
    private URI websocketServerUri;

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @Override
    public void run(ApplicationArguments args) throws Exception {
        log.info("Connecting to {}", websocketServerUri);
        new ReactorNettyWebSocketClient()
                .execute(websocketServerUri,
                        session ->
                                session.receive()
                                        .map(WebSocketMessage::getPayloadAsText)
                                        .doOnNext(log::info)
                                        .doOnTerminate(() -> log.debug("OnTerminate"))
                                        .doOnError(throwable -> log.error("OnError", throwable))
                                        .then()
                )
                .subscribeOn(Schedulers.single())
                .block();

    }
}

File: blockchain-info-aggregation-stream-parent/blockchain-info-aggregation-stream-client-example/src/main/java/info/blockchain/aggregation/client/Application.kt
package info.blockchain.aggregation.client

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.web.reactive.socket.client.ReactorNettyWebSocketClient
import reactor.core.scheduler.Schedulers
import java.net.URI

@SpringBootApplication
class Application : ApplicationRunner {

    @Value("#{new java.net.URI('\${websocket-server.uri}')}")
    private val websocketServerUri: URI? = null

    @Throws(Exception::class)
    override fun run(args: ApplicationArguments) {
        log.info("Connecting to {}", websocketServerUri)
        ReactorNettyWebSocketClient()
                .execute(websocketServerUri!!
                ) { session ->
                    session.receive()
                            .map<String> { it.payloadAsText }
                            .doOnNext { log.info(it) }
                            .doOnTerminate { log.debug("OnTerminate") }
                            .doOnError { throwable -> log.error("OnError", throwable) }
                            .then()
                }
                .subscribeOn(Schedulers.single())
                .block()

    }

    companion object {
        private val log = LoggerFactory.getLogger(Application::class.java.name)

        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(Application::class.java, *args)
        }
    }
}
